### PR TITLE
[FEATURE] [MER-938] Improve admin sections view

### PR DESCRIPTION
**Pull request by mcass19**
[MER-938](https://eliterate.atlassian.net/browse/MER-938)

Improve the admin sections index so they are able to find them easier (specially if there a lot).
 
Changes include:
- Added some columns
- Sort by all columns
- Search by some more columns
- Filter by status, type or active by date

image

Any feedback/suggestion/thought is welcome 😄 

**NOTE: This PR will be marked as "Ready for Review" and can be merged once it has 2 approvals.**

### CODE REVIEW

**Review by jrissler — APPROVED** [review]
This looks great and increases usefulness a great deal. Do you have any performance concerns?

Also does it look / work ok on mobile?

Otherwise great job :)

**Comment by mcass19** [comment]
In 69cc677712fa567346fdebe00308e3c6ec8e3b0b:
- Added some indexes to improve joins and filtering
- Digging a little bit more on how to perform full text searches, ended using `ts` based on [`postgreSQL docs`](https://www.postgresql.org/docs/current/textsearch-tables.html)
  - Added trgm indexes for searchable fields
  
@jrissler what do you think?

**Review by gastonabella — COMMENTED** [review]
Great job! 👏

File: lib/oli_web/live/common/paged_table.ex [hunk 1]
```diff
@@ -12,23 +12,24 @@ defmodule OliWeb.Common.PagedTable do
...
-      {#else}
-        <p>None exist</p>
-      {/if}
+      <div>
+        {#if @filter != ""}
+          <strong>Results filtered on &quot;{@filter}&quot;</strong>
+        {/if}
+
+        {#if @total_count > 0}
+          <Paging id="header_paging" total_count={@total_count} offset={@offset} limit={@limit} click={@page_change}/>
```
[the remark this answers is not in the document]
Comment (mcass19) [reply lib/oli_web/live/common/paged_table.ex:25]: I think I didn't understand you. If results are less than `limit` we will have only one page.

File: lib/oli_web/live/sections/sections_view.ex [hunk 2]
```diff
@@ -120,28 +106,73 @@ defmodule OliWeb.Sections.SectionsView do
...
+        <FilterBox
+          card_header_text="Browse Course Sections"
+          card_body_text=""
+          table_model={@table_model}
+          show_sort={false}
+          show_more_opts={true}>
+          <TextSearch id="text-search" text={@options.text_search}/>
+
+          <:extra_opts>
+            <Check checked={@options.active_date} click="active_date">Only active by date</Check>
```
[the remark this answers is not in the document]
Comment (mcass19) [reply lib/oli_web/live/sections/sections_view.ex:123]: haha agree 😄

File: lib/oli_web/live/common/paged_table.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcass19) [reply lib/oli_web/live/common/paged_table.ex:25]: Yeah went in that direction to simplify and be consistent with the `Listing` component which is used in some other places and do the same. But maybe it's better to not show it when we have only one page (and we can update in both)

**Review by gastonabella — APPROVED** [review]
LGTM

**Comment by jrissler** [comment]
> In [69cc677](https://github.com/Simon-Initiative/oli-torus/commit/69cc677712fa567346fdebe00308e3c6ec8e3b0b):
> 
> * Added some indexes to improve joins and filtering
> * Digging a little bit more on how to perform full text searches, ended using `ts` based on [postgreSQL docs](https://www.postgresql.org/docs/current/textsearch-tables.html)
>   
>   * Added trgm indexes for searchable fields
> 
> @jrissler what do you think?

Looks good to me - that should help :) Great job

**Pull request merged**